internal/stores/gcs.go

```diff
@@ -63,9 +63,12 @@ func (s *GCSStore) GetOptions() Options {
 	return s.Options
 }
 
-// TODO(discentem): #34 largely copy-pasted from stores/local/local.go. Can be consolidated
-// Upload generates the metadata, writes it to disk and uploads the file to the GCS bucket
+// Upload generates the metadata, writes it s.fsys and uploads the file to the GCS bucket
 func (s *GCSStore) Upload(ctx context.Context, objects ...string) error {
+	if s.Options.MetadataFileExtension == "" {
+		return ErrMetadataFileExtensionEmpty
+	}
+
```

Comment:
Yep, opinionated code here...its a `cfile` because this is a byproduct of using ccavorite itself. 

If they want to override, lets add that to the `opts` but as a follow-up

Reply:
@natewalck I filed https://github.com/discentem/cavorite/issues/106 to track this issue of duplicated code in each store provider.